edgelet/iotedge/src/check/checks/aziot_edged_version.rs

```diff
@@ -164,16 +149,67 @@ impl AziotEdgedVersion {
             .get(1)
             .expect("unreachable: regex defines one capturing group")
             .as_str();
-        self.actual_version = Some(version.to_owned());
+        Ok(version.to_owned())
+    }
+
+    async fn inner_execute(&mut self, check: &mut Check) -> anyhow::Result<CheckResult> {
+        let actual_version = self.get_installed_version(check).await?;
+        let expected_version = if let Some(expected_aziot_edged_version) =
+            &check.expected_aziot_edged_version
+        {
+            expected_aziot_edged_version.clone()
+        } else {
+            if check.parent_hostname.is_some() {
+                // This is a nested Edge device so it may not be able to access aka.ms or github.com.
+                // In the best case the request would be blocked immediately, but in the worst case it may take a long time to time out.
+                // The user didn't provide the `expected_aziot_edged_version` param on the CLI, so we just ignore this check.
+                return Ok(CheckResult::Ignored);
+            }
+
+            let versions: Vec<String> = self
+                .get_latest_released_versions(check)
+                .await?
+                .channels
+                .iter()
+                .flat_map(|channel| channel.products.iter())
+                .filter(|product| product.id == "aziot-edge")
+                .flat_map(|product| product.components.iter())
+                .filter(|component| component.name == "aziot-identity-service")
+                .map(|component| component.version.clone())
+                .collect();
+            let actual_channel = Version::parse(&actual_version)
+                .context("could not parse actual version as semver")?;
+            let expected_version = versions
+                    .iter()
+                    .find(|version| {
+                        let expected_channel = Version::parse(version)
+                            .context("could not parse expected version as semver")
+                            .unwrap(); // TODO: What's the right error handling here?
```

Comment:
Should this be `.unwrap_or_default()` instead?

Reply:
Oh, thanks for catching that! I fixed the same thing in Azure/iot-identity-service#609...I need to apply it here as well.

Reply:
Fixed.